src/controllers/sdex/index.ts

```diff
@@ -29,12 +37,38 @@ router.get(
   }),
 );
 
-router.get('/user_pool_positions', async (req, res) => {
-  res.status(200).json({
-    data: {
-      user_pool_positions: [],
-    },
-  });
-});
+router.get(
+  '/user_pool_positions',
+  asyncRoute(async (req, res) => {
+    const { userId } = req.query;
+
+    if (!userId || typeof userId !== 'string') {
+      return res.status(400).json({ error: 'User ID is required' });
+    }
+
+    const variables = {
+      user: userId,
+    };
+
+    try {
+      const response = await req.network.sdex.queryFromSubgraph<LiquidityChangesResponse>(
+        LIQUIDITY_CHANGES_QUERY,
+        variables,
+      );
+      const userPoolPositions = response.liquidityChanges;
+      const aggregatedPositions = aggregatePositions(userPoolPositions);
+
+      return maybeCacheResponse(
+        res,
+        `sdex/user_pool_positions/${req.network.chainId}/${userId}`,
+        async () => aggregatedPositions,
+        DEFAULT_CACHE_TTL,
+      ).then((data) => res.json(toResponse(data)));
```

Comment:
Put data fetching and computating inside of `maybeChacheResponse` callback, not just response:

```suggestion
      return maybeCacheResponse(
        res,
        `sdex/user_pool_positions/${req.network.chainId}/${userId}`,
        async () =>{
            const response = await req.network.sdex.queryFromSubgraph<LiquidityChangesResponse>(
            LIQUIDITY_CHANGES_QUERY,
            variables,
          );
          const userPoolPositions = response.liquidityChanges;
          const aggregatedPositions = aggregatePositions(userPoolPositions);
          return aggregatedPositions;
        },
        DEFAULT_CACHE_TTL,
      ).then((data) => res.json(toResponse(data)));
```

It will make sure that fetching and computation would be ran only once per DEFAULT_CACHE_TTL

Also, you can remove caching while developing (or at least changing TTL to small value like 1 second)